### DISCUSSION TITLE
referential stability of useQuery result

### DISCUSSION THREAD

**Discussion by lubelski** [TanStack/query#1985] [Q&A] [answered]
Apologies if this is right in the docs somewhere.  

Is the `result` from a `result = useQuery(...)` designed to be referentially stable in the case that nothing has changed about the query itself ? 

ex: can I put a query into the deps array of a `useMemo` ? or will it just be a new object on every render ?

**Comment by TkDodo** [comment, marked as answer]
It will be a new object since 3.9 because of concurrent mode support. `data` returned from useQuery will be referentially stable so you can depend on that. 

We talked here about adding that to the docs: https://github.com/tannerlinsley/react-query/issues/1905#issuecomment-801454263
